services/metadata_service/api/step.py

```diff
@@ -144,12 +144,16 @@ async def create_step(self, request):
         run_number = request.match_info.get("run_number")
         step_name = request.match_info.get("step_name")
 
-        body = await read_body(request.content)
+        body = await request.json()
         user = body.get("user_name", "")
         tags = body.get("tags")
         system_tags = body.get("system_tags")
 
-        run_number, run_id = await self._db.get_run_ids(flow_name, run_number)
+        run = await self._db.get_run_ids(flow_name, run_number)
+        if run.response_code != 200:
+            return DBResponse(400, {"message": "need to register run_id first"})
```

Comment:
Steps have a foreign key for runs at least in the V3 schema, so consider handling error code 409 specifically for this instead? 

Is it necessary to rewrite the error messages on the response handler level, or could this be handled at a higher level in `aiopg_exception_handling` instead?

Reply:
I am reworking this and yes, plan to raise an exception all the way through. The issue before though was that it would be silent.